graph-gateway/src/network/internal.rs

```diff
@@ -387,61 +387,33 @@ fn try_into_internal_indexer_info(
 
 /// Convert from the fetched subgraph information into the internal representation.
 ///
-/// The subgraph must have at least one valid deployment to be considered valid. And each
-/// deployment must have a valid manifest and at least one allocation to be considered valid.
-///
-/// If the subgraph is invalid, e.g., no valid deployments found, an error is returned.
+/// If the subgraph is invalid, e.g., has no versions, an error is returned.
 fn try_into_internal_subgraph_info(
     subgraph: subgraph::types::fetch_subgraphs::Subgraph,
 ) -> anyhow::Result<SubgraphInfo> {
     let versions = subgraph
         .versions
         .into_iter()
-        .filter_map(|version| {
+        .map(|version| {
             let deployment = version.subgraph_deployment;
 
-            let _span = tracing::trace_span!(
-                "subgraph-versions",
-                subgraph = %subgraph.id,
-                version.number = %version.version,
-                version.deployment = %deployment.id,
-            )
-            .entered();
-
-            // Deployment must have a valid manifest to be considered valid.
-            let (deployment_manifest_network, deployment_manifest_start_block) = match deployment
+            let deployment_manifest_network = deployment
                 .manifest
-            {
-                // Deployment must have a valid manifest to be considered valid.
-                None => {
-                    tracing::trace!("filtering-out version-deployment: no manifest");
-                    return None;
-                }
-                Some(manifest) if manifest.network.is_none() => {
-                    tracing::trace!("filtering-out version-deployment: no manifest network info");
-                    return None;
-                }
-                Some(manifest) => (manifest.network?, manifest.start_block.unwrap_or(0)),
-            };
+                .as_ref()
+                .and_then(|manifest| manifest.network.clone());
+            let deployment_manifest_start_block = deployment
+                .manifest
+                .as_ref()
+                .and_then(|manifest| manifest.start_block);
```

Comment:
could avoid the string clone here
```rust
            let deployment_manifest_start_block = deployment
                .manifest
                .as_ref()
                .and_then(|manifest| manifest.start_block);
            let deployment_manifest_network =
                deployment.manifest.and_then(|manifest| manifest.network);
```

Reply:
Noted down. As I am overhauling this part to include errors, I will avoid that 👍🏼